shell/browser/file_system_access/file_system_access_permission_context.cc

```diff
@@ -0,0 +1,889 @@
+    base::FilePath::StringType hostname) {
+  return !base::ranges::all_of(hostname, [](base::FilePath::CharType c) {
+    return (c >= L'A' && c <= L'Z') || (c >= L'a' && c <= L'z') ||
+           (c >= L'0' && c <= L'9') || (c == L'.') || (c == L'-');
+  });
+}
+
+bool MaybeIsLocalUNCPath(const base::FilePath& path) {
+  if (!path.IsNetwork()) {
+    return false;
+  }
+
+  const std::vector<base::FilePath::StringType> components =
+      path.GetComponents();
+
+  // Check for server name that could represent a local system. We only
+  // check for a very short list, as it is impossible to cover all different
+  // variants on Windows.
+  if (components.size() >= 2 &&
+      (base::FilePath::CompareEqualIgnoreCase(components[1],
+                                              FILE_PATH_LITERAL("localhost")) ||
+       components[1] == FILE_PATH_LITERAL("127.0.0.1") ||
+       components[1] == FILE_PATH_LITERAL(".") ||
+       components[1] == FILE_PATH_LITERAL("?") ||
+       ContainsInvalidDNSCharacter(components[1]))) {
+    return true;
+  }
+
+  // In case we missed the server name check above, we also check for shares
+  // ending with '$' as they represent pre-defined shares, including the local
+  // drives.
+  for (size_t i = 2; i < components.size(); ++i) {
+    if (components[i].back() == L'$') {
+      return true;
+    }
+  }
+
+  return false;
+}
+#endif
+
+// Sentinel used to indicate that no PathService key is specified for a path in
+// the struct below.
+constexpr const int kNoBasePathKey = -1;
+
+enum class BlockType {
+  kBlockAllChildren,
+  kBlockNestedDirectories,
+  kDontBlockChildren
+};
+
+const struct {
+  // base::BasePathKey value (or one of the platform specific extensions to it)
+  // for a path that should be blocked. Specify kNoBasePathKey if |path| should
+  // be used instead.
+  int base_path_key;
+
+  // Explicit path to block instead of using |base_path_key|. Set to nullptr to
+  // use |base_path_key| on its own. If both |base_path_key| and |path| are set,
+  // |path| is treated relative to the path |base_path_key| resolves to.
+  const base::FilePath::CharType* path;
+
+  // If this is set to kDontBlockChildren, only the given path and its parents
+  // are blocked. If this is set to kBlockAllChildren, all children of the given
+  // path are blocked as well. Finally if this is set to kBlockNestedDirectories
+  // access is allowed to individual files in the directory, but nested
+  // directories are still blocked.
+  // The BlockType of the nearest ancestor of a path to check is what ultimately
+  // determines if a path is blocked or not. If a blocked path is a descendent
+  // of another blocked path, then it may override the child-blocking policy of
+  // its ancestor. For example, if /home blocks all children, but
+  // /home/downloads does not, then /home/downloads/file.ext will *not* be
+  // blocked.
+  BlockType type;
+} kBlockedPaths[] = {
```

Comment:
This list seems kind of brittle, can we somehow reference the upstream one?

Reply:
@MarshallOfSound not that I could determine - we could patch somewhere I guess?

Reply:
Yeah I think maybe patching https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/file_system_access/chrome_file_system_access_permission_context.cc would be better than copy-pasting this list into e/e 🤔 The risk of drift here isn't "something breaks" it's "the fs api is now unsafe in electron" which is a pretty bad consequence

Reply:
01fe40e